### PR TITLE
add devtool utilities

### PR DESCRIPTION
**Pull request by fmigneault**
## Changes

* Add `LICENSE` file.
* Add `bump-my-version` with `make version` and `make VERSION=<...> bump` utilities to self-update release versions.
* Add more metadata to `pyproject.toml`.
* Adjust `README.md` with updated references and release version indicators.
* Add `CHANGES.md` to record version updates.
* Add `dev` dependencies to `pyproject.toml` for testing the package (install with `pip install ".[dev]"`).
* Remove `requirements.txt` in favor of all dependencies combined in `pyproject.toml`.
* Add test to validate STAC Collection and Item contain `source` with expected THREDDS format.
* Fix broken tests and invalid imports.

### CODE REVIEW

File: CHANGES.md
Comment (dchandan) [inline CHANGES.md:13]: This bullet point seems not relevant to this PR.

**Review by dchandan — APPROVED** [review]
Looks good. Thanks for adding these!

**Review by huard — APPROVED** [review]

File: CHANGES.md
Comment (fmigneault) [reply CHANGES.md:13]: Yes. This is because the Changelog did not already exist. It would be present following the merge of yesterday. I can separate it from this PR into distinct versions to make it clearer.

Comment (fmigneault) [reply CHANGES.md:17]: Done in https://github.com/crim-ca/stac-populator/pull/28/commits/40c34cdc896fae2c84fa056ea4ac254c41720a4c

**Comment by dchandan** [comment]
Shall we merge this?

**Comment by fmigneault** [comment]
@dchandan 
> Shall we merge this?

I'm working on the CI to make sure it runs the test successfully. 
Will merge right after it does.

**Comment by fmigneault** [comment]
@dchandan @huard 
Everything works except python 3.12 that needs https://github.com/xarray-contrib/xncml/pull/57

File: tests/data/stac_collection_testdata_xclim_cmip6_catalog.json
Comment (dchandan) [inline tests/data/stac_collection_testdata_xclim_cmip6_catalog.json:1]: Why are there no `properties` in this STAC item?

**Review by dchandan — CHANGES_REQUESTED** [review]

File: STACpopulator/populator_base.py
Comment (fmigneault) [reply STACpopulator/populator_base.py:57]: To bypass the config loading strategy that assumes a file as needed. It is not mandatory to make a populator work if arguments can be provided by some other mean, such as CLI args.

File: tests/data/stac_collection_testdata_xclim_cmip6_catalog.json
Comment (fmigneault) [reply tests/data/stac_collection_testdata_xclim_cmip6_catalog.json:1]: This is a STAC Collection. There should never be `properties`.

File: STACpopulator/populator_base.py
Comment (fmigneault) [reply STACpopulator/populator_base.py:128]: The test shows how it can be useful to decouple the logic. 
It is possible this way for an implementation to do something else with the created STAC Collection data before publishing it, such as adding more links, assets or metadata only relevant to the collection and not the items.

File: tests/data/stac_collection_testdata_xclim_cmip6_catalog.json
Comment (dchandan) [reply tests/data/stac_collection_testdata_xclim_cmip6_catalog.json:1]: Oh right, this is a collection..

File: STACpopulator/populator_base.py
Comment (dchandan) [reply STACpopulator/populator_base.py:128]: OK

File: STACpopulator/api_requests.py
Comment (fmigneault) [reply STACpopulator/api_requests.py:1]: Sure. I don't mind either way because it doesn't actually stops someone from modifying it. It does create a lot of "mismatching" types however if we start to be strict with mypy and IDE linters, which is why I modified it. We would need to add multiple `cast` to ensure types are correct.

**Review by dchandan — APPROVED** [review]

**Pull request merged**
